app/controllers/donations_controller.rb

```diff
@@ -0,0 +1,48 @@
+class DonationsController < ApplicationController
+  def new
+    @donation = Donation.new
+  end
+
+  def create
+    @donation = Donation.new(donation_params)
+
+    token = params[:stripeToken]
+
+    customer = Stripe::Customer.create(
+      :email        => @donation.email,
+      :card         => token,
+      :description  => "Donation of $#{@donation.total} to Operation Code",
+    )
+
+    charge = Stripe::Charge.create(
+      :customer     => customer.id,
+      :amount       => @donation.total,
+      :description  => "Donation of $#{@donation.total} to Operation Code",
+      :currency     => 'usd'
+    )
+
+    respond_to do |format|
+      if @donation.save
+        DonationMailer.thankyou(@donation).deliver_now
+        format.html { redirect_to root_url, 
+                      :flash => { :success => "Thank you for your generous donation!"}
+                    }
+      else
+        format.html { redirect_to new_donation_path,
```

Comment:
Avoid using {...} for multi-line blocks.<br>Block body expression is on the same line as the block start.